### PR TITLE
Added Transactions and Essential vs Non-Essential Use Cases.

### PR DESCRIPTION
**Pull request by Divnoor-C**
Added Transactions and Essential vs Non-Essential Use Cases.

### CODE REVIEW

**Comment by Divnoor-C** [comment]
Also made sure no errors, but when you push make sure you review that email verification is how it should be

File: src/main/java/osiris/LocalDateAdapter.java
Comment (Mahan-H) [inline src/main/java/osiris/LocalDateAdapter.java:1]: consider moving this to test resources packages if possible

File: src/main/java/osiris/OffsetDateTimeAdapter.java
Comment (Mahan-H) [inline src/main/java/osiris/OffsetDateTimeAdapter.java:1]: consider moving this to test resources packages if possible

File: src/main/java/osiris/data_access/DBUserDataAccessObject.java
Comment (Mahan-H) [inline src/main/java/osiris/data_access/DBUserDataAccessObject.java:1]: this class should also implement GrabTransactionUserDataAccessInterface

File: src/main/java/osiris/OffsetDateTimeAdapter.java
Comment (Divnoor-C) [reply src/main/java/osiris/OffsetDateTimeAdapter.java:1]: Just realized I didnt need it

**Pull request merged**